packages/ckeditor5-table/src/tableproperties/tablepropertiesediting.js

```diff
@@ -242,6 +242,12 @@ function enableTableToFigureProperty( schema, conversion, options ) {
 	schema.extend( 'table', {
 		allowAttributes: [ modelAttribute ]
 	} );
-	upcastStyleToAttribute( conversion, { viewElement: /^(table|figure)$/, ...options } );
+
+	upcastStyleToAttribute( conversion, {
+		viewElement: /^(table|figure)$/,
+		isApplicable: element => element.name != 'table' || element.parent.name != 'figure',
```

Comment:
```suggestion
		isApplicable: element => !( element.name == 'table' && element.parent.name == 'figure' ),
```

this way it's more readable IMHO.